File: src/main/java/com/junction/tonight/spark/domain/ZepMap.java
package com.junction.tonight.spark.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "zep_map", catalog = "zep")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ZepMap {

    @Id
    @Column(name = "map_hash")
    private String mapHash;

    @Column
    private String mapName;

    @Column
    @ColumnDefault(value = "0")
    private Integer visitedCnt;
}

File: src/main/java/com/tonight/spark/domain/kotlin/ZepMap.kt
package com.tonight.spark.domain.kotlin

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity(name = "zep_map")
data class ZepMap(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @Column(name = "map_hash")
        var mapHash: String? = null,

        @Column(name = "map_name")
        var mapName: String? = null
){

}